src/core/frames/frame_controller.js

```diff
@@ -114,6 +115,10 @@ export class FrameController {
   }
 
   async #loadSourceURL() {
+    if (this.#isFrameRefreshMorph()) {
+      this.#prepareFrameToMorph()
+    }
```

Comment:
Right before a frame is about to be updated, we check to see if it's supposed to morph.